### PR TITLE
Adding a contributing page.

### PR DESCRIPTION
**Pull request by sfshaza2**
I had to close the old PR, it became a giant mess.

@Hixie, this incorporates your feedback. Yours too, @chalin. 

Please review ASAP. I don't want to move all the docs under /docs until I can merge this.

### CODE REVIEW

File: src/docs.md [hunk 1]
```diff
@@ -47,26 +46,6 @@ Some of the new content includes:
 If you have questions or comments about the revamped site, [file an
 issue]({{site.repo.this}}/issues).
 
-{% comment %}
-# TODO (Not for publication)
```
Comment (chalin) [inline src/docs.md]: 👍

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+---
+title: Contribute to Flutter docs
+description: Ways to contribute to the Flutter project.
+---
```
[the remark this answers is not in the document]
Comment (chalin) [reply CONTRIBUTING.md]: I'll let @Hixie cover the rest of the content.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+If you would like to help contribute to the Flutter project,
+we’re happy to have your help! Anyone can contribute, whether you’re new to
+the project or you’ve been around a long time, and whether you self-identify
+as a developer, an end user, or someone who just can’t stand seeing typos.
+
+For more ways to get involved in the Flutter community or to learn about us,
+also visit the [Flutter community](/community) page.
+
+Looking for the [Effective Dart]({{site.www}}/guides/language/effective-dart)?
```
Comment (Hixie) [inline CONTRIBUTING.md]: i don't really understand why this paragraph about effective dart is on this page. It seems weird to link to the Dart style guide but not, say, the Flutter style guide. (Not that either really make sense to link to from this doc as I see it, but maybe I'm missing something?)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+[issues]: https://github.com/flutter/flutter/issues
+[doc-issues]: https://github.com/flutter/website/issues
+[so]: https://stackoverflow.com/tags/flutter
+[mailinglist]: https://groups.google.com/d/forum/flutter-dev
+[gitter]: https://gitter.im/flutter/flutter
+[reddit]: https://www.reddit.com/r/FlutterDev
+
+For more information on contributing, see
+[Contributing to Flutter](https://github.com/flutter/flutter/blob/master/CONTRIBUTING.md)
+in the [Flutter SDK](https://github.com/flutter/flutter) repo.
```
Comment (Hixie) [inline CONTRIBUTING.md]: this final paragraph is what I would have expected to see near the top. This is the main way I hope to see people contribute: by writing code and pushing PRs.

**Comment by sfshaza2** [comment]
Closed and created a new PR, incorporating this feedback.

**Comment by sfshaza2** [comment]
New PR: https://github.com/flutter/website/pull/1605

**Pull request closed without merging**